isbn_updater/src/comp.php

```diff
@@ -0,0 +1,352 @@
+		fwrite($myfile, "ERROR: There was an error updating the ISBN" ."\t\r\n");
+	} else {
+		echo "<br/> - ISBN Updated Successfully to ". var_export($new_isbn13s, true) ."</br>";
+		fwrite($myfile, "ISBN Updated Successfully" ."\t\r\n");
+	}
+
+}
+
+/**
+ * Get the Item that we need to edit
+ *
+ * @param  mixed $shortCode The tenant short code
+ * @param  mixed $item_id The item Id to fetch
+ * @param  mixed $TalisGUID Your talis GUID
+ * @param  mixed $token Your access token for the API.
+ * @return boolean|object
+ */
+function getItem($shortCode, $item_id, $TalisGUID, $token) {
+	$url = "https://rl.talis.com/3/$shortCode/draft_items/$item_id?include=resource.part_of" ;
+
+	$ch = curl_init();
+	curl_setopt($ch, CURLOPT_URL, $url);
+	curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+	curl_setopt($ch, CURLOPT_HTTPHEADER, array(
+		
+		"X-Effective-User: $TalisGUID",
+		"Authorization: Bearer $token",
+		'Cache-Control: no-cache'
+	
+	));
+	$output = curl_exec($ch);
+	$info = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+	// echo $info;
+	$output_json = json_decode($output);
+	curl_close($ch);
+	
+	if ($info !== 200){
+		echo "<p>ERROR: There was an error getting the resource:</p><pre>" . var_export($output, true) . "</pre>";
+		return false;
+	} else {
+		// echo "Resource details acquired </br>";
+		// var_export($output);
+		return json_decode($output);
+	}
+}
+
+/**
+ * Look at an item's data and detect the resources in it.
+ *
+ * @param  mixed $item The JSON API response from the get item request
+ * @return array[mixed] an array of key value pairs that describe the primary and secondary resources.
+ */
+function detect_resources($item){
+	$output = [
+		'primary' => false,
+		'primary_id' => false,
+		'secondary' => false,
+		'secondary_id' => false
+	];
+
+	$resources = array_filter($item->included, function($var){return $var->type == 'resources';});
+	var_export(count($resources));
+	if (count($resources) == 2){
+		// there are two resources
+		foreach($resources as $resource){
+			// one of the resources has a relationship to another resource
+			if (!empty($resource->relationships->part_of)){
+				$output['primary_id'] = $resource->id;
+				$output['primary'] = $resource;
+				$output['secondary_id'] = $resource->relationships->part_of->data->id;
+				$output['secondary'] = array_filter($resources, function($var) use ($output){return $var->id == $output['secondary_id'];})[0];
+			} 
+		}
+	} else {
+		// there is just the one resource
+		$output ['primary_id'] = $resources[0]->id;
+		$output ['primary'] = $resources[0];
+	}
+	// var_export($output);
+	return $output;
+}
+
+$token = getToken($clientID, $secret);
+
+//***********Running Code******************
+$file_handle = fopen($uploadfile, "rb");
+
+while (!feof($file_handle) )  {
+	echo "<br />----------- <br />";
+
+	$line_of_text = fgets($file_handle);
+	$parts = explode(",", $line_of_text);
+
```

Comment:
Defensive check:

I think it might be worth checking that `count($parts === 3)` at this point.  Currently it's fine without as we are checking that [1] and [2] are valid ISBNS, and [0] will fail to return an item if it is wrong.  But if this is extended in the future with other functionality it could end up catching some bugs where the csv file is malformed.

Reply:
Implemented